cuappdev/announcements-backend
# BACKEND-3: Add Jest testing

## Overview

Add Jest testing for methods in `AppService` and `AnnouncementService` classes.

## Changes Made

### Testing

- Create a `tests` folder outside of the `src` folder that contains testing code.
- Create a `mocks` folder inside of the `tests` folder that contains factories to create mock objects used for testing.
- Mocks are created using the `faker` library.
- Create `tests/utils/dbConnection.ts` to manage DB connection. Note that MongoDB memory server is used as the testing DB.

### Other

- Configure a GitHub Actions workflow that triggers the testing suite on PR and merge to `main`.
- Update `README.md` with instructions for testing.
- Update `AnnouncementService.updateAnnouncement` to validate dates.
- Update `AppService.validateAppSlugs` to return a boolean for testing purposes.

## Test Coverage

- Ran unit tests using `yarn test`.
